test/test.js

```diff
@@ -884,6 +884,182 @@ describe('Page', function() {
     });
   });
 
+  describe('Frame.waitForXpath', function() {
+    const addElement = tag => document.body.appendChild(document.createElement(tag));
+
+    it('should immediately resolve promise if node exists', async({page, server}) => {
```

Comment:
If we share the same predicate code with waitForSelector, I think a lot of these tests become redundant.

Reply:
I've changed the code to be shared. Which ones do you think we should keep?

Reply:
Okay I've used my best judgement and removed most of them.